runtime/protocol/http-client-engines/http-client-engine-ktor/common/src/aws/smithy/kotlin/runtime/http/engine/ktor/KtorEngine.kt

```diff
@@ -61,11 +61,9 @@ public class KtorEngine(
         followRedirects = false
     }
 
-    private val logger = Logger.getLogger<KtorEngine>()
-
     override suspend fun roundTrip(context: ExecutionContext, request: HttpRequest): HttpCall {
         val callContext = callContext()
-        val reqLogger = logger.withContext(context)
+        val reqLogger = context.getLogger<KtorEngine>()
```

Comment:
question: Why even keep the concept of logger around if we are centralizing around trace spans?

Reply:
`Logger` seemed like a nice abstraction that captured the notion of a `TraceSpan` contextualized with a source component. But that said, I don't think it's strictly _necessary_.

With the code that exists now, we can do things like apply a source component to a span and use the combination repeatedly:

```kotlin
val logger = context.getLogger<MyClass>()
logger.info { "Foo" }
logger.debug { "Bar" }
logger.trace { "Baz" }
```

Without the notion of a logger, we could use extension methods of `TraceSpan` itself and pass in the component every time:

```kotlin
context.traceSpan.info<MyClass> { "Foo" }
context.traceSpan.debug<MyClass> { "Bar" }
context.traceSpan.trace<MyClass> { "Baz" }
```

This might even be more efficient since it skips the instantiation of a wrapper class, the overhead of which is inversely proportional to the number of logging statements. I don't necessarily love the source component boilerplate but it's an alternative worth considering.

Reply:
I see because it's not creating a child span it's utilizing source component...🤔 

I suppose that makes sense. I think my concern is that if we want tracing to be what everything goes through then my gut says don't have a mixed system where we use one or the other. It will (as it has in this PR) inevitably result in some things going through logger and some going through tracing (and then to logger). It also adds a decision point that need not exist: "should I use the logger or tracing API?". 

That being said I agree it seems like we would want extensions to enable ergonomic use (although the syntax you showed isn't the worst). Would be good to discuss this more with the team.